src/main/java/com/powsybl/openloadflow/sa/AbstractSecurityAnalysis.java

```diff
@@ -262,38 +261,56 @@ SecurityAnalysisResult runSimulationsOnAllComponents(LfNetworkList networks, Lis
+        for (LfNetwork n : networkToSimulate) {
+            SecurityAnalysisResult resultOtherComponent = runSimulations(n, propagatedContingencies, parameters, securityAnalysisParameters, operatorStrategies, actions, limitReductions);
 
-                // PostContingency and OperatorStrategies results
-                mergeSecurityAnalysisResult(resultOtherComponent, postContingencyResultMap, operatorStrategyResultMap, n.getNumCC());
-            }
-            postContingencyResults = postContingencyResultMap.values().stream().toList();
-            operatorStrategyResults = operatorStrategyResultMap.values().stream().toList();
+            // Merge into first result
+            // PreContingency results first
+            preContingenctViolations.addAll(resultOtherComponent.getPreContingencyResult().getLimitViolationsResult().getLimitViolations());
+            mergedPreContingencyNetworkResult = mergeNetworkResult(mergedPreContingencyNetworkResult, resultOtherComponent.getPreContingencyResult().getNetworkResult());
+
+            // PostContingency and OperatorStrategies results
+            mergeSecurityAnalysisResult(resultOtherComponent, postContingencyResultMap, operatorStrategyResultMap, n.getNumCC());
         }
+        postContingencyResults = postContingencyResultMap.values().stream().toList();
+        operatorStrategyResults = operatorStrategyResultMap.values().stream().toList();
+
         PreContingencyResult mergedPrecontingencyResult =
                 new PreContingencyResult(result.getPreContingencyResult().getStatus(),
                         new LimitViolationsResult(preContingenctViolations),
                         mergedPreContingencyNetworkResult);
         return new SecurityAnalysisResult(mergedPrecontingencyResult, postContingencyResults, operatorStrategyResults);
     }
 
+    static Optional<LfNetwork> selectValidMainComponent(LfNetworkList networks) {
+        return networks.getList().stream().filter(n -> isMainCCandMainSC(n) && n.getValidity().equals(LfNetwork.Validity.VALID)).findFirst();
+    }
+
+    static List<LfNetwork> selectValidSecondaryComponents(LfNetworkList networks, LoadFlowParameters.ConnectedComponentMode mode) {
+        return networks.getList().stream()
+            .filter(n -> (mode == LoadFlowParameters.ConnectedComponentMode.ALL && !isMainCCandMainSC(n)
+                || mode == LoadFlowParameters.ConnectedComponentMode.MAIN && isMainCCandSecondarySC(n))
+                && n.getValidity().equals(LfNetwork.Validity.VALID)).toList();
+    }
+
+    static boolean isMainCCandMainSC(LfNetwork n) {
+        return n.getNumCC() == ComponentConstants.MAIN_NUM && n.getNumSC() == ComponentConstants.MAIN_NUM;
+    }
+
+    static boolean isMainCCandSecondarySC(LfNetwork n) {
+        return n.getNumCC() == ComponentConstants.MAIN_NUM && n.getNumSC() != ComponentConstants.MAIN_NUM;
+    }
+
```

Comment:
`AbstractSecurityAnalysis` doesn't seem to be the right place for these methods, move them to `LfNetworkList` ?